opts/mount.go

```diff
@@ -161,6 +187,15 @@ func (m *MountOpt) Set(value string) error {
 		return fmt.Errorf("cannot mix 'tmpfs-*' options with mount type '%s'", mount.Type)
 	}
 
+	if mount.BindOptions != nil && mount.BindOptions.ReadOnlyForceRecursive {
```

Comment:
This probably needs a check for `ReadOnlyNonRecursive` as well, as `readonly=false,bind-recursive=readonly` makes no sense.

Reply:
> as `readonly=false,bind-recursive=readonly`

Ah, yes, that was the case I mentioned above (and was hoping it would work, as I could see use-cases for a "writable" bind-mount with _nested mounts_ read-only), but currently not a combination that was possible; https://github.com/docker/cli/pull/4316#discussion_r1340348214

Reply:
Just to be sure; in the current implementation, does `--mount type=bind,readonly,.....` do the same as `-v source:target:ro` (so best-effort recursive), or do we need to make changes to the `readonly` option to make that happen (I lost track again on which combination did that, and what parts we did on the daemon side)

Reply:
> This probably needs a check for `ReadOnlyNonRecursive` as well, as `readonly=false,bind-recursive=readonly` makes no sense.

Done (I assume you meant `readonly=false,bind-recursive=writable`)

Reply:
> Just to be sure; in the current implementation, does `--mount type=bind,readonly,.....` do the same as `-v source:target:ro` (so best-effort recursive), or do we need to make changes to the `readonly` option to make that happen (I lost track again on which combination did that, and what parts we did on the daemon side)

The former one.